blog/2022-05-13-caching-api-requests-with-long-lived-workflows.md

```diff
@@ -0,0 +1,231 @@
+  const res = await axios.get('https://cdn.moneyconvert.net/api/latest.json');
+  return res.data.rates;
+};
+```
+
+Next, let's write a Workflow that calls this Activity once per day.
+With Temporal Workflows, you can simply write a `while (true) {}` loop with a [JavaScript sleep](https://masteringjs.io/tutorials/fundamentals/sleep) that pauses the Workflow until the next time you need to refresh exchange rates.
+Writing the Workflow this way may seem counterintuitive, because we've all had to learn over the course of our careers that writing applications isn't this easy.
+But, with Temporal, it actually is this easy!
+
+```ts
+import {
+  defineQuery,
+  proxyActivities,
+  setHandler,
+  sleep
+} from '@temporalio/workflow';
+import type * as activities from './activities';
+
+const { getExchangeRates } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1 minute',
+});
+
+export const getExchangeRatesQuery = defineQuery<any, [string]>('getExchangeRates');
+
+export async function exchangeRatesWorkflow(): Promise<any> {
+  let rates: any = null;
+
+  // Register a query handler that allows querying for the current rates
+  setHandler(getExchangeRatesQuery, () => rates);
+
+  while (true) {
+    // Get the latest rates
+    rates = await getExchangeRates();
+
+    // Sleep until tomorrow at 12pm server time, and then get the rates again
+    const today = new Date();
+    const tomorrow = new Date(today);
+    tomorrow.setHours(12, 0, 0, 0);
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    // @ts-ignore
+    await sleep(tomorrow - today);
+  }
+}
+```
+
+That's the entire Workflow for caching exchange rates.
+The full source code is available in [this GitHub repo](https://github.com/vkarpov15/temporal-api-caching-example).
+Notice there's no explicit references to a database or job queue.
+This Workflow is almost pure business logic, with a minimum of references to frameworks or services.
+
+To run this Workflow, you can run a `start-workflow.ts` script as shown below.
+This script starts a Workflow and exits, leaving the Workflow running on the Worker.
+Note that only one Workflow with a given `workflowId` can run at any given time, so the below code also ensures that only one copy of this Workflow is running at any given time.
+
+```ts
+import { WorkflowClient } from '@temporalio/client';
+import { exchangeRatesWorkflow } from '../workflows';
+
+run().catch((err) => {
+  console.error(err);
+  process.exit(1);
+});
+
+async function run() {
+  const client = new WorkflowClient();
+
+  const handle = await client.start(exchangeRatesWorkflow, {
+    taskQueue: 'exchange-rates',
+    workflowId: 'exchange-rates-workflow',
+  });
+  console.log(`Started workflow ${handle.workflowId}`);
+}
+```
+
+Note that there is one key detail that this Workflow is missing: a [Continue As New](https://docs.temporal.io/docs/concepts/what-is-continue-as-new).
+There's more about that later in this blog post.
```

Comment:
I'm stetting this instance of "there's" because it's quite conversational. (Blog posts get a lot more leeway than documentation.)